WordPress/gutenberg
# New block: Add a Breadcrumbs block

## Description


Related to #21943

This PR proposes adding in a new Breadcrumbs block, to render navigation links of the hierarchy of a page. This is a server rendered block, which would typically be used in a page template, most likely before or after the Post Title.

## Features include

* Markup follows this example: https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/examples/breadcrumb/
* Support for pages (hierarchical post types) _or_
* Support for displaying a single hierarchical category as breadcrumbs for posts
* Set a custom separator character (click on the separator in the editor canvas to edit)
* Show leading separator (so that the breadcrumb can be rendered like a back button)
* Show current page / post title
* Show site title
* Custom override text to override the site title (click on the site title in the editor canvas to edit)


Some of these features are implemented in a subtle way so as to not be intrusive. To override the site title, once it's displayed as the first breadcrumb, a user can click into the site title and manually update it. Likewise, the separator character defaults to `/` but can be edited by clicking on it and replacing it with another character.



## Testing instructions





1. Use a block-based theme such as TT3
2. Open up the site editor and edit a template part that will be used for pages (or any hierarchical post type)
3. Insert the Breadcrumbs block to the template part so that it's rendered when you view a page on the front end of the site
4. Note that in the site editor, the block will display placeholder names for each of the breadcrumbs
5. Create a couple of pages, and assign parents to these pages so that you have nested pages on your site
6. View the page on the front end of the site, and you should see the Breadcrumbs rendered in the appropriate place in the template

Repeat the above with a normal post, and use hierarchical post types.

If you wish to test the custom block spacing control, be sure to use an FSE theme, and set the `settings.spacing.blockGap` value to `true` in your `theme.json` file (for more information on this see the testing instructions in a blockGap PR like #34630)

## Screenshots 



![image](https://github.com/WordPress/gutenberg/assets/14988353/c1d85b8f-2692-4de4-ae75-b7c77683cffd)

https://github.com/WordPress/gutenberg/assets/14988353/92e15e00-ab79-4582-8886-a75f852af496
